수정사항

hyb26 opened the pull request:
각 커밋의 body에도 동일한 내용 적혀있습니다!

- Comment : @table 제거, @Setter 추가
- CommentController : 공백 및 주석 수정
- CommentDto : Response에 reviewUuid 추가
- CommentMapper : @mapping 추가
- CommentRepository : query문 작성
- CommentServiceImpl : 로직 작성 및 @transactional 수정

- Notification : @table 제거, @Setter 추가, join문 수정
- NotificationController : 공백 및 로직 수정
- NotificationDto : Post 삭제
- NotificationMapper : dto 수정에 따른 수정.
- NotificationRepository : query문(findByUserUuid, findAllByIsRead, countByYetRead) 작성
- NotificationServiceImpl : 로직 작성 및 @transactional 수정, method 수정 및 추가
- NotificationService : 로직 추가에 따른 수정

- Place : @table 제거, @Setter 삭제 및 추가, hide 추가
- PlaceImg, Heart : @table 제거, @Setter 삭제
- PlaceController : 로직 추가 및 수정
- PlaceDto : Response 수정
- PlaceMapper : 이름 수정 및 메서드 추가.
- PlaceRepository : query문 작성
- PlaceServiceImpl :  @transactional 수정, method 수정 및 추가
- PlaceService : 로직 추가에 따른 수정

hyb26 committed:
:sparkles: [Feat] Comment 관련 Controller, Service, Repository, Mapper 추가 (작성, 전체조회, 삭제, 상세 조회)

hyb26 committed:
:sparkles: [Feat] Notification 관련 Controller, Service, Repository, Mapper 추가 (전체조회, 읽음처리, 알림 갯수 반환)

hyb26 committed:
:sparkles: [Feat] Place 관련 entity(place, heart, placeimg), dto 수정 및 추가

hyb26 committed:
:sparkles: [Feat] Place 관련 entity(place, heart, placeimg), dto, controller 수정, Comment 및 Notification 수정

hyb26 committed:
:sparkles: [Feat] 수정사항
- 피드백 주신 부분들 수정
- comment, heart, notification, place, placeimg의 entity id명 수정
- comment, notification controller에서 ResponseEntity로 return하지 않은 부분들 중 일부 수정 및 나머지 로직은 반환값 고민중(금요일 전까지 수정 예정)
- CommentServiceImpl, NotificationServiceImpl에서 @Transactional(readOnly) 추가

hyb26 committed:
:sparkles: [Feat] 수정사항
- 피드백 주신 부분들 수정 및 comment의 getComment -> verifiedByUuid로 변경.

hyb26 committed:
:sparkles: [Feat] Comment 관련 파일들 전체 수정
- Comment : @Table 제거, @Setter 추가
- CommentController : 공백 및 주석 수정
- CommentDto : Response에 reviewUuid 추가
- CommentMapper : @Mapping 추가
- CommentRepository : query문 작성
- CommentServiceImpl : 로직 작성 및 @Transactional 수정

hyb26 committed:
:sparkles: [Feat] Notification 관련 파일들 전체 수정
- Notification : @Table 제거, @Setter 추가, join문 수정
- NotificationController : 공백 및 로직 수정
- NotificationDto : Post 삭제
- NotificationMapper : dto 수정에 따른 수정.
- NotificationRepository : query문(findByUserUuid, findAllByIsRead, countByYetRead) 작성
- NotificationServiceImpl : 로직 작성 및 @Transactional 수정, method 수정 및 추가
- NotificationService : 로직 추가에 따른 수정

hyb26 committed:
:sparkles: [Feat] Place 관련 파일들 전체 수정
- Place : @Table 제거, @Setter 삭제 및 추가, hide 추가
- PlaceImg, Heart : @Table 제거, @Setter 삭제
- PlaceController : 로직 추가 및 수정
- PlaceDto : Response 수정
- PlaceMapper : 이름 수정 및 메서드 추가.
- PlaceRepository : query문 작성
- PlaceServiceImpl :  @Transactional 수정, method 수정 및 추가
- NotificationService : 로직 추가에 따른 수정

seungchul-lee118 commented on sogo/src/main/java/com/enzinior/sogo/comment/entity/Comment.java:
```diff
@@ -0,0 +1,44 @@
+    private long commentId;
+
+    @Column(name = "comment_uuid", columnDefinition = "varchar(80)", unique = true)
+    private String uuid = UUID.randomUUID().toString();
```
각 엔티티에 uuid로 할지 [엔티티]+Uuid 로 할건지 논의 필요

hyb26 replied:
확인했습니다!

seungchul-lee118 commented on sogo/src/main/java/com/enzinior/sogo/comment/mapper/CommentMapper.java:
```diff
@@ -0,0 +1,25 @@
+    @Mapping(source = "reviewUuid", target = "review.uuid")
+    @Mapping(source = "userUuid", target = "user.uuid")
+    @Mapping(source = "parentUuid", target = "comment.uuid")
+    Comment commentPostToComment(CommentDto.Post requestBody);
```
CommentDto에서 comment의 uuid를 commentUuid로 받았기 때문에
Comment 엔티티의 필드명을 uuid -> commentUuid 변경 혹은 Mapper에 commentUuid, uuid 매핑 필요

hyb26 replied:
확인했습니다. 엔티티 필드명을 일단 바꿔둘게요!

seungchul-lee118 commented on sogo/src/main/java/com/enzinior/sogo/comment/service/CommentServiceImpl.java:
```diff
@@ -0,0 +1,71 @@
+            comment.setSecret(false);
+        }else{
+            comment.setSecret(true);
+        }
```
comment.setSecret(!comment.isSecret()) 으로 코드 단축 가능

hyb26 replied:
와 그러네요..!! 수정하겠습니다! 감사합니다ㅎㅎ

seungchul-lee118 commented on sogo/src/main/java/com/enzinior/sogo/notification/entity/Notification.java:
```diff
@@ -0,0 +1,41 @@
+    private long notificationId;
+
+    @Column(name = "notification_uuid", columnDefinition = "varchar(80)", unique = true)
+    private String uuid;
```
Comment처럼 다환이랑 셋이서 uuid 네이밍 정리 필요

hyb26 replied:
넵 확인했습니다!

seungchul-lee118 commented on sogo/src/main/java/com/enzinior/sogo/notification/entity/Notification.java:
```diff
@@ -0,0 +1,41 @@
+    @JoinColumn(name = "review_id")
+    @Setter
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Review review;
```
알림이 Review랑 연관관계가 있는건가..?
무조건 리뷰에 대한 알림인가?
우리 신고때도 알림 가지 않나?

hyb26 replied:
리뷰에 대한 알림이 들어올 경우에 이미지와 해당 리뷰의 uuid를 반환해야해서 넣어뒀어요! 알림 리스트를 반환할 때 결국 리뷰에 대한 것이든 아니면 신고나 첫 손님에 대한 알림이든 같이 반환해야해서 response 타입을 동일하게 만든다고 했을 때 review에 대해 연관관계를 맺어두고 null이 가능하도록 하면 되지 않을까 싶어서 넣어뒀었습니다.. 제가 찾아본 바로는 joincolumn은 null값을 가질 수 있다고 찾았었는데 제가 잘못알고 있는 걸까요?
혹은 더 좋은 방법이 있을까요?!!

seungchul-lee118 replied:
ManyToOne 으로 관계설정을 해두면 Table에서 FK 설정이 되어버려서 무조건 리뷰에 대한 정보를 가지고 있어야한다고 알고 있었는데
가능하다고 하네 ㅎㅎ

https://stackoverflow.com/questions/25718229/can-a-manytoone-jpa-relation-be-null

hyb26 replied:
오오 확인 감사합니다ㅎㅎ

hyb26 committed:
:sparkles: [Feat] uuid 관련 항목들 수정
- commentUuid
- notificationUuid
- placeUuid
- 조언주신대로 로직 리팩터링 (!comment.isSecret())
